File: Samples/CS/CodeSamples/SubmitMultiSample.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading.Tasks;
using Inetlab.SMPP;
using Inetlab.SMPP.Common;
using Inetlab.SMPP.PDU;

namespace CodeSamples
{
    public class SubmitMultiSample
    {
        readonly SmppClient _client = new SmppClient();

        public async Task SendToMultipleRecepients()
        {
            //<SendToMultipleRecepients>
           await _client.Submit(SMS.ForSubmitMulti()
                .ServiceType("test")
                .Text("Test Test")
                .From("MyService")
                .To("1111")
                .To("2222")
                .To("3333")
            );

            //</SendToMultipleRecepients>
        }

        public void SendToPhoneNumbers(List<string> phoneNumbers)
        {
            //<SendToPhoneNumbers>
            var pduBuilder = SMS.ForSubmitMulti()
                .ServiceType("test")
                .Text("Test Test")
                .From("MyService");

            foreach (string phoneNumber in phoneNumbers)
            {
                pduBuilder.To(phoneNumber);
            }

            //</SendToPhoneNumbers>
        }

        public async Task SendToDestinationList()
        {
            //<SendToDestinationList>
            List<IAddress> destList = new List<IAddress>();

            destList.Add(new SmeAddress("11111111111", AddressTON.Unknown, AddressNPI.ISDN));
            destList.Add(new DistributionList("my_destribution_list_on_SMPP_Server"));

            var submitResponses = await _client.Submit(SMS.ForSubmitMulti()
                .ServiceType("test")
                .Text("Test Test")
                .From("MyService")
                .ToDestinations(destList)
            );
            //</SendToDestinationList>
        }
    }
}

File: Samples/VB.NET/CodeSamples/SubmitMultiSample.vb
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports System.Threading.Tasks
Imports Inetlab.SMPP
Imports Inetlab.SMPP.Common
Imports Inetlab.SMPP.PDU

Namespace CodeSamples
	Public Class SubmitMultiSample
		Private ReadOnly _client As New SmppClient()

		Public Async Function SendToMultipleRecepients() As Task
			'<SendToMultipleRecepients>
		   Await _client.Submit(SMS.ForSubmitMulti().ServiceType("test").Text("Test Test").From("MyService").To("1111").To("2222").To("3333"))

			'</SendToMultipleRecepients>
		End Function

		Public Sub SendToPhoneNumbers(ByVal phoneNumbers As List(Of String))
			'<SendToPhoneNumbers>
			Dim pduBuilder = SMS.ForSubmitMulti().ServiceType("test").Text("Test Test").From("MyService")

			For Each phoneNumber As String In phoneNumbers
				pduBuilder.To(phoneNumber)
			Next phoneNumber

			'</SendToPhoneNumbers>
		End Sub

		Public Async Function SendToDestinationList() As Task
			'<SendToDestinationList>
			Dim destList As New List(Of IAddress)()

			destList.Add(New SmeAddress("11111111111", AddressTON.Unknown, AddressNPI.ISDN))
			destList.Add(New DistributionList("my_destribution_list_on_SMPP_Server"))

			Dim submitResponses = Await _client.Submit(SMS.ForSubmitMulti().ServiceType("test").Text("Test Test").From("MyService").ToDestinations(destList))
			'</SendToDestinationList>
		End Function
	End Class
End Namespace
